### PR TITLE
fix: Improved HEAD request fail test when fallback-ing to GET request

### PR DESCRIPTION
**Pull request by Robloche**
This should close https://github.com/shaka-project/shaka-player/issues/6016.

Following PR https://github.com/shaka-project/shaka-player/pull/5986, it appears that servers don't always return correct HTTP status error code when `HEAD` request is not supported. It should be `405` but badly configured servers could return `404`.

So I replaced:

```
if (error && error.code == shaka.util.Error.Code.HTTP_ERROR) {
  request.method = 'GET';
  ...
}
```

by:

```
if (error && (error.code == shaka.util.Error.Code.HTTP_ERROR || error.code == shaka.util.Error.Code.BAD_HTTP_STATUS)) {
  request.method = 'GET';
  ...
}
```

This change was made in `lib/hls/hls_parser.js` and `lib/net/networking_utils.js`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,5 +6,6 @@ docs/api/
 coverage/
 .DS_Store
 .vscode
+.idea
```
Comment (avelad) [inline .gitignore:9]: Please revert these changes

File: lib/text/ui_text_displayer.js [hunk 2]
```diff
@@ -574,6 +574,36 @@ shaka.text.UITextDisplayer = class {
     return Cue.positionAlign.CENTER;
   }
 
+  /**
```
Comment (avelad) [inline lib/text/ui_text_displayer.js:577]: Please revert these changes

**Review by avelad — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (Robloche) [reply .gitignore]: Done.

File: lib/text/ui_text_displayer.js (hunk 2 above)
Comment (Robloche) [reply lib/text/ui_text_displayer.js]: Done. I might have messed up with my branch...

File: .gitignore [hunk 3]
```diff
@@ -7,4 +7,4 @@ coverage/
 .DS_Store
 .vscode
 .babel-cache
-.idea/
```
Comment (avelad) [inline .gitignore:10]: Please revert the changes of this file.

**Review by avelad — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (Robloche) [reply .gitignore:10]: Done. I'm so bad with git. That makes me cry.

Comment (avelad) [reply .gitignore:10]: recheck please, you removed one line in this file.

**Review by avelad — CHANGES_REQUESTED** [review]
Revert the changes of .gitignore

File: .gitignore (hunk 3 above)
Comment (Robloche) [reply .gitignore:10]: Well... Yes, I removed `.idea` in a commit, then I realised that I forgot a line and removed `/_mytest/` in a second commit.
Those are 2 folders I use in my IDE.

**Review by avelad — APPROVED** [review]
